stubs/redis/redis/sentinel.pyi

```diff
@@ -39,5 +42,5 @@ class Sentinel(object):
     def discover_master(self, service_name): ...
     def filter_slaves(self, slaves): ...
     def discover_slaves(self, service_name): ...
-    def master_for(self, service_name, redis_class=..., connection_pool_class=..., **kwargs) -> None: ...
-    def slave_for(self, service_name, redis_class=..., connection_pool_class=..., **kwargs) -> None: ...
+    def master_for(self, service_name, redis_class=..., connection_pool_class=..., **kwargs) -> Redis[_StrType]: ...
```

Comment:
We don't need to make the class generic as the return type is localized to the methods. Instead, we could use overloads like this (untested):

```python

_R = TypeVar("_R", bound=Redis[Any])

class Sentinel:
...
    @overload
    def master_for(self, service_name, redis_class: type[_R] = ..., connection_pool_class=..., **kwargs) -> Redis[_R]: ...
    @overload
    def master_for(self, service_name, *, connection_pool_class=..., **kwargs) -> Redis[str]: ...
```

Reply:
This should work:

```python
_R = TypeVar("_R", bound=Redis[Any])

class Sentinel:
...
    @overload
    def master_for(self, service_name, redis_class: type[_R], connection_pool_class=..., **kwargs) -> _R: ...
    @overload
    def master_for(self, service_name, *, connection_pool_class=..., **kwargs) -> Redis[str]: ...
```

So we have two cases:
1. If `redis_class` is given, that is used as the return type.
2. If `redis_class` is not given, we return `Redis[str]`.

In the second overload, `connection_pool_class` cannot be a positional argument, because `obj.master_for(service_name, ConnectionPool)` passes the `ConnectionPool` as the `redis_class` argument, not as the `connection_pool_class` argument. So we set that to keyword only with `*` before it.

Reply:
Unfortunately, `type[_R]` is not working in Python versions < 3.9 :-/

Reply:
mypy 0.910 (the latest version) is buggy in this regard. We need to use `typing.Type` still.